### DISCUSSION TITLE
Unable to get correct direction of flow when using Atmospheric Boundary Layer (ABL) condition.

### DISCUSSION THREAD

**Discussion by akshaymani94** [nextfoam/baram#127] [Q&A]
![Screenshot from 2024-10-15 17-55-15](https://github.com/user-attachments/assets/d298fd9e-4d12-4b90-adfe-a3a9d33c3b82)
The above highlighted patch is the inlet of my domain. I want to specify an ABL boundary condition on this face. The details given in the ABL boundary condition are given below.
![Screenshot from 2024-10-15 17-50-51](https://github.com/user-attachments/assets/497f89fa-87dc-4110-98a6-47c6a2dd2cda)
The outlet of the domain is shown below. It is specified as pressure outlet.
![Screenshot from 2024-10-15 17-55-29](https://github.com/user-attachments/assets/14d8ee76-4019-4b7c-81b6-f74c279a896e)
![Screenshot from 2024-10-15 17-51-21](https://github.com/user-attachments/assets/f8690ab5-9e7d-47fa-b3a2-7581ef5d4ae7)

When I specify this boundary condition, the flow is seen to be occurring from outlet to inlet. I want it to actually occur in the opposite direction. Can you tell me the mistake that I might be committing here leading to this error. I also tried changing the normal vector, but even that did not give me any fruitful results. 
When I specify a velocity inlet boundary condition with velocity of 5m/s, I am getting the correct direction. Can anybody help me solve this issue.

**Comment by kdg5424** [comment]
Hello @akshaymani94. This is Donggyu from NEXTFOAM.
There are several suspicious things.

1) Could you let me know the other BCs? For example, how did you configure BC on the left side and right side?
![image](https://github.com/user-attachments/assets/fd3b5340-feea-4080-a66d-7ad6cc63dc2a)
2) How about the residuals? Is it ok residuals?
3) Could you check your flow direction again? Is the below picture right?
![image](https://github.com/user-attachments/assets/ade96284-0b8e-4e65-bd88-75f44030db34)

**Reply by akshaymani94** [reply]
![Screenshot from 2024-10-16 10-06-06](https://github.com/user-attachments/assets/25539e45-145a-4442-9d09-7423efa15c13)
The side and top walls were specified initially as walls, but then I tried specifying all the three as symmetry boundary condition. The residuals and the flow direction in this case are shown below.
![Screenshot from 2024-10-16 09-55-32](https://github.com/user-attachments/assets/879e0f93-b6e0-4b64-a3a8-b8d3a02085a0)
![Screenshot from 2024-10-16 09-51-10](https://github.com/user-attachments/assets/c2b0d7b3-ed09-4cca-ae0a-945f5f42c9a9)

At the time of initialization the flow looks something like this.
![Screenshot from 2024-10-16 10-14-43](https://github.com/user-attachments/assets/43d88dfa-aeb4-4fde-8a6f-369fcb11f907)

**Comment by kdg5424** [comment]
Hello @akshaymani94.
Could you send your case file here?
Please attatch your case file after initialization.
Then I'll check your case file

Thanks

Best regards

Donggyu Kim

**Reply by kdg5424** [reply]
Furthermore, the symmetry BC looks like weird. (It's just my opinion, I could be wrong)
I'll discuss with the other engineers.

**Reply by akshaymani94** [reply]
HI Sir,
Thankyou for your response. The case file is around 4.3GB in size. Can I share you a dropbox link.

**Reply by kdg5424** [reply]
Wow... The size is too large.
Could you share it after initilaization from copy? The size would be smaller than 4.3GB.
If 4.3GB is the size after initialization, then just send your dropbox link.
Thanks.

**Reply by akshaymani94** [reply]
Sorry, I couldnt upload it on dropbox either.
Could it be the symmetry boundary condition that is causing the issue. This is just an external flow simulation in a 500m*300m*100m domain. Could you suggest any other method to solve this issue.

**Reply by kdg5424** [reply]
I think yes it could be.
If the flow direction is straight forward, you can use symmetry boundary condition of course.
But your flow direction is not straight forward and it has the angle.
Unfortunately, I'm not sure why your problem happens and also not sure the problem happens because of symmetry boundary condition without your case.
So maybe I can test with simple case but my workload is not enough now. So it will take some time to give you the reason and solutions.

If you're ok, I'll give you the reply next week. Is it ok?

Best regards

Donggyu Kim

**Comment by kdg5424** [comment]
Hello @akshaymani94.
Although the simulation've not over yet, I share my result.
It is the initial state.
![image](https://github.com/user-attachments/assets/655d49ba-a02c-4422-91f0-eb3cd2a12b87)

![image](https://github.com/user-attachments/assets/9ac0ead9-5603-45c9-b4b6-f81cbd7d4b1e)
And it's the vector field after 200 iterations.

![image](https://github.com/user-attachments/assets/4310dcc0-910e-4a62-be39-dff2d7135448)
And my residuals is like this.

I think symmetry BC could be the problem, so I suggest you these three things.

**1) I suggest you to rotate your mesh. So the flow could be straght forward. And simulate again.**
I suggest you this strongly.
You can rotate your mesh from here : 
![image](https://github.com/user-attachments/assets/8f6d67df-70fb-4d9e-97d2-2e3a31f78bd8)


2) Could you check one more thing?
Have you configure wall as ABL wall?
![image](https://github.com/user-attachments/assets/66c36416-276a-4c4a-b176-4a84480c7ae6)
You can change like this.

3) Could you check your pressure contour from the center height?

Best regards

Donggyu Kim

**Reply by akshaymani94** [reply]
Hi @kdg5424,
I cannot thank you enough for taking the time to solve my query. I changed my symmetry boundary condition back to the wall boundary condition. I also tried using the outflow boundary condition in place of the pressure outlet boundary condition at the exit. This corrected the flow direction from inlet to outlet, but there was still some flow exiting through the inlet. 
![Screenshot from 2024-10-18 10-34-06](https://github.com/user-attachments/assets/d3ef8e76-d132-484b-bd49-7311e4835a4a)
I was unaware about setting the wall as ABL wall. I will try doing this from now on. Also thank you for suggesting the option for rotating the mesh. I will try both these suggestions. I am sure I have a solution now. Thanks a lot Donggyu Kim for your help.

Thanks and regards
Akshay

**Reply by kdg5424** [reply]
Hello @akshaymani94. This is Donggyu from NEXTFOAM.

I'm very glad to hear that you solve your problem.
But still there are some concerns for BCs.
If you change symmetry to the wall boundary condition, the velocity will not be calculated precisely around there.
Anyway, I hope you decide it well and make a fine result

Best regards

Donggyu Kim
